app/controllers/flavor_controller.rb

```diff
@@ -10,15 +10,59 @@ class FlavorController < ApplicationController
   include Mixins::GenericSessionMixin
 
   def self.display_methods
-    %w(instances)
+    %w[instances]
+  end
+
+  def new
+    assert_privileges('flavor_create')
+    drop_breadcrumb(:name => _("Add a new Flavor"), :url => "/flavor/new")
+    @in_a_form = true
+    @id = 'new'
+  end
+
+  def button
+    if params[:pressed] == 'flavor_create'
+      javascript_redirect(:action => 'new')
+    else
+      delete_flavors
+    end
+  end
+
+  def ems_list
+    assert_privileges('flavor_create')
+    ems_list = ManageIQ::Providers::CloudManager.select do |ems|
+      ems.class::Flavor.supports?(:create)
+    end
+    ems_list.each do |ems|
+      {:name => ems.name, :id => ems.id}
+    end
+    render :json => {:ems_list => ems_list}
   end
 
   private
 
   def textual_group_list
-    [%i(properties relationships), %i(tags)]
+    [%i[properties relationships], %i[tags]]
   end
   helper_method :textual_group_list
 
+  def delete_flavors
+    assert_privileges('flavor_delete')
+    checked = find_checked_items
+    checked[0] = params[:id] if checked.blank? && params[:id]
+    FlavorController.model.where(:id => checked).each do |flavor|
```

Comment:
This should be `find_records_with_rbac(model, checked_or_params)`, and you don't need to do the `checked[0] = params[:id]` etc..